docs/howtos/DataPlane Generation - DPG/08methodInputs.mdx

```diff
@@ -0,0 +1,509 @@
+
+```typescript
+```
+
+</TabItem>
+<TabItem value="java" label="Java" >
+
+```java
+```
+
+</TabItem>
+<TabItem value="go" label="Go" >
+
+```go
+```
+
+</TabItem>
+</Tabs>
+
+### HTTP Post
+
+<Tabs>
+<TabItem value="typespec" label="TypeSpec" default>
+
+```typespec
+op @post post(User): void;
+```
+
+</TabItem>
+
+<TabItem value="python" label="Python">
+
+For model inputs, Python automatically generates an overload that allows SDK users to input through a file stream.
+
+TODO: generating this many method signatures feels verbose
+
+```python
+@overload
+def post(user: User, *, content_type: str = "application/json") -> None:
+  ...
+
+@overload
+def post(user: JSON, *, content_type: str = "application/json") -> None:
+  ...
+
+@overload
+def post(user: IO[bytes], *, content_type: str = "application/json") -> None:
+  ...
+
+def post(user: [User, JSON, IO[bytes]], *, content_type: str = "application/json") -> None:
+  ...
+```
+
+</TabItem>
+<TabItem value="csharp" label="CSharp" >
+
+```csharp
+```
+
+</TabItem>
+<TabItem value="typescript" label="Typescript" >
+
+```typescript
+```
+
+</TabItem>
+<TabItem value="java" label="Java" >
+
+```java
+```
+
+</TabItem>
+<TabItem value="go" label="Go" >
+
+```go
+```
+
+</TabItem>
+</Tabs>
+
+## Spread
+
+### Named model
+
+<Tabs>
+<TabItem value="typespec" label="TypeSpec" default>
+
+```typespec
+model User {
+  firstName: string;
+  lastName: string;
+}
+
+op upload(...User): void;
+```
+
+</TabItem>
+
+<TabItem value="python" label="Python">
+
+For Python, we will also generate the overloads described in the Http Post section, but omitting for brevity
+
+```python
+class User:
+  first_name: str
+  last_name: str
+
+def upload(user: User) -> None:
+  ...
+```
+
+</TabItem>
+<TabItem value="csharp" label="CSharp" >
+
+```csharp
+```
+
+</TabItem>
+<TabItem value="typescript" label="Typescript" >
+
+```typescript
+```
+
+</TabItem>
+<TabItem value="java" label="Java" >
+
+```java
+```
+
+</TabItem>
+<TabItem value="go" label="Go" >
+
+```go
+```
+
+</TabItem>
+</Tabs>
+
+### Anonymous Model
+
+<Tabs>
+<TabItem value="typespec" label="TypeSpec" default>
+
+```typespec
+op upload(firstName: string, lastName: string): void;
+```
+
+</TabItem>
+
+<TabItem value="python" label="Python">
+
+For Python, we will also generate the overloads described in the Http Post section, but omitting for brevity
+
+```python
+def upload(first_name: str, last_name: str) -> None:
+  ...
+```
+
+</TabItem>
+<TabItem value="csharp" label="CSharp" >
+
+```csharp
+```
+
+</TabItem>
+<TabItem value="typescript" label="Typescript" >
+
+```typescript
+```
+
+</TabItem>
+<TabItem value="java" label="Java" >
+
+```java
+```
+
+</TabItem>
+<TabItem value="go" label="Go" >
+
+```go
+```
+
+</TabItem>
+</Tabs>
+
+### Alias
+
+<Tabs>
+<TabItem value="typespec" label="TypeSpec" default>
+
+```typespec
+alias User = {
+  firstName: string,
+  lastName: string,
+};
+
+op upload(...User): void;
```

Comment:
for me, it is same as anonymous model spread. i wonder can we arrange the cases with more essential way. i mean:
1. `op update(param1, param2): void` raw parameter without decorator
2. `op update(@path path1, param1): void` raw parameters with decorator
3. `op update(...ModelWithoutDecoratedProperties): void` spread model without decorated properties
4. `op update(...ModelWithDecoratedProperties): void` spread model with decorated properties
5. `op update(...Alias/AnonymouModelWith/WithoutDecoratedProperties): void` spread anonymous model with/without decorated properties

Reply:
i'll switch the raw parameters to alias bc you are right, they are the same